flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/aggfunctions/AvgAggFunction.scala

```diff
@@ -137,6 +146,15 @@ abstract class BigIntegralAvgAggFunction[T] extends AggregateFunction[T] {
     }
   }
 
+  override def retract(accumulator: Accumulator, value: Any): Unit = {
```

Comment:
If I directly create a LongAvgAggFunction, it will be difficult to return null for avg result, as null.asInstanceOf[Long] will return 0.